### PR TITLE
cal site updated scraper is failing

### LINKED ISSUE #763: Cal site updated, scraper is failing
Cal site has new design, and the scraper made for it is failing. 
Any intention to fix it?

### PR DESCRIPTION
**Pull request by baruchiro**
Fixes #763 

New Cal website support.

### CODE REVIEW

File: src/scrapers/visa-cal.ts [hunk 1]
```diff
@@ -45,6 +47,94 @@ interface ScrapedAdditionalInfo {
...
+        }[];
+      }[];
+      immidiateDebits: { totalDebits: [], debitDays: [] };
+    }[];
+    blockedCardInd: boolean;
+  };
+  statusCode: 1;
+  statusDescription: string;
+  statusTitle: string;
+}
```
Comment (baruchiro) [inline src/scrapers/visa-cal.ts:136]: @erikash I will appreciate your help here.
I managed to fetch the transactions per month, and now I need to convert their object to our `Transaction` object.
I hate this phase of understanding all their fields and validating it with the content I see on the website.

Can you help me with that?

Thanks!!

Comment (erikash) [reply src/scrapers/visa-cal.ts:136]: Hi @baruchiro 
Sorry but I only just saw your message - sure will do this weekend

Comment (erikash) [reply src/scrapers/visa-cal.ts:136]: @baruchiro I've converted the transactions here:
https://github.com/eshaham/israeli-bank-scrapers/pull/769

Tested it a little with installments, regular transactions and multiple accounts.

**Comment by erikash** [comment]
Just to update on the status - 
it's currently working but it's really rough and requires refactoring before merging

**Comment by baruchiro** [comment]
@erikash I merged your changes from #769 into here.

**Comment by erikash** [comment]
@baruchiro I'm working on additional improvements - I've already discovered some bugs and will push some commits later today

**Comment by baruchiro** [comment]
@eshaham @esakal PR is ready.

Refactor is welcome but I think we can move on.

**Review by erikash — APPROVED** [review]
Looks like a good baseline - @itairaz1 feel free to extend this with categories (behind a flag) in a separate PR

**Comment by baruchiro** [comment]
@eshaham @esakal ping

**Comment by orentet** [comment]
when will this be merged?

**Comment by esakal** [comment]
@baruchiro  @erikash Excellent work! It appears that this task was quite challenging! I am now merging it.

**Pull request merged**